crates/bevy_state/src/state_scoped.rs

```diff
@@ -91,3 +91,74 @@ pub fn clear_state_scoped_entities<S: States>(
         }
     }
 }
+
+/// Entities marked with this component will be removed
+/// when the world's state of the matching type matches the supplied value.
```

Comment:
```suggestion
/// upon entering the given state.
```
Optional: Again this is copy/pasted old docs, but I think it sounded awkward before and a little more awkward in this case. Calling it "the world's state" is not technically wrong, but it's not helpful (like calling an entity "the world's entity").

The docs for `DespawnOnExitState` can also be updated to match.